solver/src/settlement_simulation.rs

```diff
@@ -1,12 +1,20 @@
 use crate::encoding::EncodedSettlement;
-use anyhow::{Context, Result};
+use anyhow::{Error, Result};
 use contracts::GPv2Settlement;
 use ethcontract::{batch::CallBatch, dyns::DynTransport, transaction::TransactionBuilder};
 use futures::FutureExt;
 use shared::Web3;
+use web3::types::{BlockId, BlockNumber};
 
 const SIMULATE_BATCH_SIZE: usize = 10;
 
+pub enum Block {
+    // Simulate the transactions at this block and attach a tenderly link to errors.
+    FixedWithTenderly(u64),
```

Comment:
nano-naming-nit: Sounds to me like the simulation is done via tenderly (but any more descriptive name like FixedWithTenderlyLinkErrorContext sounds a bit silly).

Reply:
Agree name is not perfect which is why I added the comment above. Imo good enough.